staging/src/k8s.io/client-go/discovery/cached/disk/cached_discovery.go

```diff
@@ -84,11 +84,6 @@ func (d *CachedDiscoveryClient) ServerResourcesForGroupVersion(groupVersion stri
 		klog.V(3).Infof("skipped caching discovery info due to %v", err)
 		return liveResources, err
 	}
-	if liveResources == nil || len(liveResources.APIResources) == 0 {
```

Comment:
caching `nil` seems wrong, so I wouldn't drop that early return case.

If we get a non-nil result, does that always mean we got a well-formed but empty APIResourceList back from the server here:

https://github.com/kubernetes/kubernetes/blob/418ae605ec1b788d43bff7ac44af66d8b669b833/staging/src/k8s.io/client-go/discovery/discovery_client.go#L384-L387

Reply:
1. What should the API return for an empty group without resources? You can see in my kubectl example that this is quite common for certain operators (i.e. metric-server).
2. What would you like to cache in thise case? We could map "nil" to some kind of "empty response" object and then map it back on load. Would be more complex though.

Reply:
> What should the API return for an empty group without resources?

I'd expect a well-formed but empty APIResourceList response

> What would you like to cache in thise case?

Caching the non-nil empty APIResourceList would make sense